### PR TITLE
Settings page

### PR DESCRIPTION
**Pull request by Boy132**
Adds a simple Settings page. Settings are directly written to the .env file.
The old admin settings stuff is removed.

This supersedes #259.

grafik
grafik
grafik
grafik
grafik

### CODE REVIEW

**Comment by Poseidon281** [comment]
You are kidding me right. After I and with the help of Lance spend all this time working on my PR. You are just going to make another?

File: app/Filament/Pages/Settings.php [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+                                        ->live()
+                                        ->columnSpanFull()
+                                        ->formatStateUsing(fn ($state): bool => (bool) $state)
+                                        ->afterStateUpdated(fn ($state, Set $set) => $set('PANEL_CLIENT_ALLOCATIONS_ENABLED', (bool) $state))
+                                        ->default(env('PANEL_CLIENT_ALLOCATIONS_ENABLED', config('panel.client_features.allocations.enabled'))),
+                                    TextInput::make('PANEL_CLIENT_ALLOCATIONS_RANGE_START')
+                                        ->label('Starting Port')
+                                        ->required(true)
+                                        ->numeric()
+                                        ->minValue(1024)
```
Comment (notAreYouScared) [inline app/Filament/Pages/Settings.php:236]: i think this has to be 1025, as the ports have to be > 1024? i might be wrong

Comment (Boy132) [reply app/Filament/Pages/Settings.php:236]: No, ports below 1024 are restricted. So 1 - 1023 isn't allowed.

Comment (notAreYouScared) [reply app/Filament/Pages/Settings.php:236]: Correct. Greater than 1024. So the min would be 1025?

Comment (parkervcp) [reply app/Filament/Pages/Settings.php:236]: Below 1024 is 1-1023. The usable range is 1024 and above.

Comment (RMartinOscar) [reply app/Filament/Pages/Settings.php:236]: Ports in an allocation must be greater than 1024 and less than or equal to 65535.
image

**Comment by RMartinOscar** [comment]
Why fully removing the `APP_ENVIRONMENT_ONLY` instead of setting the default to false ?
I find it very useful for troubleshooting

**Comment by Boy132** [comment]
> Why fully removing the `APP_ENVIRONMENT_ONLY` instead of setting the default to false ? I find it very useful for troubleshooting

Because it's no longer doing anything. Settings are directly written to the `.env` file.

**Comment by RMartinOscar** [comment]
> > Why fully removing the `APP_ENVIRONMENT_ONLY` instead of setting the default to false ? I find it very useful for troubleshooting
> 
> Because it's no longer doing anything. Settings are directly written to the `.env` file.

Mybad thought it was still saved in the settings table of the db

File: app/Filament/Pages/Settings.php (hunk 1 above)
Comment (Boy132) [reply app/Filament/Pages/Settings.php:236]: > Ports in an allocation must be greater than 1024 and less than or equal to 65535. image

Then we need to update that bit. It's definitely _below_ 1024, so 1024 itself is valid.

**Comment by notAreYouScared** [comment]
I feel this is a good start, overtime i can add everything we have an env value for.

**Comment by notAreYouScared** [comment]
I feel this is good enough in its current state to be added, More settings can be added along the way.

**Review by notAreYouScared — APPROVED** [review]

**Pull request merged**